src/mesh/interpolation.cxx

```diff
@@ -178,51 +190,58 @@ const Field3D interp_to(const Field3D &var, CELL_LOC loc)
 	  }
 	}
 	break;
-	// Need to communicate in Y
       }
       case CELL_ZLOW: {
-        for(const auto &i : result.region(RGN_NOZ)) {
+        for(const auto &i : result.region(region)) {
 	  s.c = var[i];
 	  s.p = var[i.zp()];
 	  s.m = var[i.zm()];
 	  s.pp = var[i.offset(0,0,2)];
 	  s.mm = var[i.offset(0,0,-2)];
 	  
-	  result[i] = interp(s);
+          if ((location == CELL_CENTRE) && (loc == CELL_ZLOW)) {
+            // Producing a stencil centred around a lower Z value
+            s.pp = s.p;
+            s.p  = s.c;
+          } else if(location == CELL_ZLOW) {
+            // Stencil centred around a cell centre
+            s.mm = s.m;
+            s.m  = s.c;
+          }
+          
+ 	  result[i] = interp(s);
 	}
 	break;
       }
       default: {
 	// This should never happen
-	throw BoutException("Don't know what to do");
+	throw BoutException("Unrecognized stagger location in interp_to()");
       }
       };
       
       if(dir != CELL_ZLOW) {
-	// COMMUNICATION
-	
-	mesh->communicate(result);
-
-	// BOUNDARIES
+        // COMMUNICATION
+        if ( region!=RGN_NOBNDRY ) fieldmesh->communicate(result);
+        // BOUNDARIES
```

Comment:
While we're touching things here, both of these comments can go.

Also, if you have clang-format installed, please run it over these bits :)

Reply:
Ran clang-format over interpolation.hxx interpolation.cxx and the new test_interpolation.cxx file.